src/server/scan/listener/helpers/process-scan-complete.js

```diff
@@ -8,28 +9,39 @@ import { config } from '~/src/config'
 
 const callbackQueue = config.get('sqsScanResultsCallback')
 async function processScanComplete(server, uploadId) {
-  const upload = await server.redis.findUploadWithFiles(uploadId)
-  if (isUploadPending(upload.uploadStatus) && isScanningComplete(upload)) {
-    upload.uploadStatus = uploadStatus.ready.description
-    upload.numberOfInfectedFiles = numberOfInfectedFiles(upload)
-    await server.redis.storeUploadDetails(uploadId, upload)
-    if (upload.scanResultCallbackUrl) {
+  const { files, uploadDetails } =
+    await server.redis.findUploadAndFiles(uploadId)
```

Comment:
so if there is no result here then this will blow up. Maybe guard this:

```
const uploadAndFilesResult = await server.redis.findUploadAndFiles(uploadId)
const files = uploadAndFilesResult?.files
const uploadDetails = uploadAndFilesResult?.uploadDetails

```